connection.js

```diff
@@ -459,6 +459,65 @@ class Connection {
         }
         /******** end of BOARDSIZES *********/
 
+        if (notification.handicap === -1 && config.noautohandicap) {
+            conn_log("no autohandicap");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
+
+        if (notification.handicap === -1 && config.noautohandicapranked && notification.ranked) {
+            conn_log("no autohandicap for ranked games");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled for ranked games on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
+
+        if (notification.handicap === -1 && config.noautohandicapunranked && !notification.ranked) {
+            conn_log("no autohandicap for unranked games");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled for unranked games on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
+
+        /***** automatic handicap min/max handicap limits detection ******/
+
+        // below is a fix of automatic handicap bypass issue
+        // by manually calculating handicap stones number
+        // then calculate if it is within set min/max
+        // limits set by botadmin
+        if (notification.handicap === -1 && !config.noautohandicap && !config.noautohandicapranked && !config.noautohandicapunranked) {
```

Comment:
IIUIC, this will reject some handicaps that were automatically determined by OGS? I would not want to enable this for my bot because I'm happy letting the players play with whatever settings they like. So you probably should add an option to enable this behavior?

Reply:
no it will not @windo , if you look : 
- from line 483 : https://github.com/online-go/gtp2ogs/pull/165/files#diff-0d74c7455b8cbe5be6f644690ecb1b8aR483
- to line 491 : https://github.com/online-go/gtp2ogs/pull/165/files#diff-0d74c7455b8cbe5be6f644690ecb1b8aR491

the only thing that that code does is reject > 9 stones only if the game is ranked and automatic handicap is allowed : on ogs server, you cant play against a ranked game against someone 9 ranks stronger than you, so if automatic handicap picks more than 9 stones in a ranked game, it doesnt make sense at all : thats all this condition wants to do @windo 

then later in the code from line 493 https://github.com/online-go/gtp2ogs/pull/165/files#diff-0d74c7455b8cbe5be6f644690ecb1b8aR493 until the end, we only reject some specific values, and everything else is allowed